Source/gg2/Objects/InGameElements/Spectator.events/End Step.xml

```diff
@@ -77,6 +77,11 @@ else if(tracking)
         }
         else
             stopTracking = true;
+        if (player.corpse and !player.object and tracking)
```

Comment:
(Reposted)
instance_create() increases a hidden global object id counter. This counter, being an unsigned int, can go into massive numbers before it has to cycle back to 0.
So no, it doesn't guarantee it (Like all counters), but its REALLY difficult to make it reuse an object ID.